### PR TITLE
DAOS-7273 pool: use target_addr_list

### PR DESCRIPTION
**Pull request by wangdi1**
It should use pool_target_addr_list in
ds_mgmt_pool_target_update_state(), since
the target index(within a rank) is being
used until real pool map change.

Signed-off-by: Di Wang <di.wang@intel.com>

### CODE REVIEW

**Comment by wangdi1** [comment]
ping

**Review by kccain — APPROVED** [review]
understand this patch goal is to reduce issues/confusion by using struct pool_target_addr_list() in this code path instead of struct pool_target_id_list()

I guess there is an additional aspect of the discussion in DAOS-7273 that is considering exclude a rank (and all targets) vs excluding all targets of a rank (but not the rank itself). I added a note/question there separately.

**Comment by wangdi1** [comment]
> understand this patch goal is to reduce issues/confusion by using struct pool_target_addr_list() in this code path instead of struct pool_target_id_list()
> 
> I guess there is an additional aspect of the discussion in DAOS-7273 that is considering exclude a rank (and all targets) vs excluding all targets of a rank (but not the rank itself). I added a note/question there separately.

File: src/mgmt/srv_pool.c [hunk 1]
```diff
@@ -433,7 +432,7 @@ ds_mgmt_pool_target_update_state(uuid_t pool_uuid, d_rank_list_t *svc_ranks,
 		 * than allocating an actual list array and populating it
 		 */
 		reint_ranks.rl_nr = 1;
-		reint_ranks.rl_ranks = &rank;
+		reint_ranks.rl_ranks = &target_addrs->pta_addrs[0].pta_rank;
```
Comment (liw) [inline src/mgmt/srv_pool.c:435]: If you refresh this patch, might be worthwhile to add a comment for this function, stating that target_addrs must only include targets belonging to just one rank.

**Review by liw — APPROVED** [review]

**Comment by mjmac** [comment]
Should this be landed for 2.0, or is it better to land after branching?

**Pull request merged**